### PR TITLE
Update Haskell tests

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by sipa** [comment]
Care to review @benma @roconnor-blockstream?

**Comment by gmaxwell** [comment]
These extra vectors were created as a result of introducing errors in the C and C++ implementations and observing that the tests still passed.  I think it might be helpful for someone to try doing something similar for the haskell implementation since a different set of bugs might be not covered by the existing tests.

File: ref/haskell/test/Spec.hs [hunk 1]
```diff
@@ -73,6 +86,10 @@ tests = testGroup "Tests"
                 let checksumEncoded = bech32Encode resultHRP resultData
                     expectedChecksum = Just $ BSC.map toLower checksum
                 assertEqual (show checksum ++ " re-encode") expectedChecksum checksumEncoded
+    , testCase "Invalid checksums" $ forM_ invalidChecksums $ \checksum -> do
+          case bech32Decode checksum of
+            Just (_, _) -> assertFailure (show checksum)
```
Comment (benma) [inline ref/haskell/test/Spec.hs]: Simpler: `Just _ -> `

Comment (roconnor-blockstream) [reply ref/haskell/test/Spec.hs]: forM_ invalidChecksums $ 
    \checksum -> assertBool (show checksum) (isNothing $ bech32Decode checksum)

Similar to `Invalid Addresses` below.

File: ref/haskell/test/Spec.hs [hunk 2]
```diff
@@ -22,6 +22,18 @@ validChecksums = map BSC.pack
     , "split1checkupstagehandshakeupstreamerranterredcaperred2y9e3w"
     ]
 
+invalidChecksums :: [BS.ByteString]
+invalidChecksums = map BSC.pack
+    [ " 1nwldj5"
+    , "\x7F" ++ "1axkwrx"
```
Comment (roconnor-blockstream) [inline ref/haskell/test/Spec.hs]: In principle this should be `"\x7F\&1axkwrx"` or `"\DEL1axkwrx"`, not that it matters too much.

**Comment by sipa** [comment]
@benma @roconnor-blockstream Thanks, updated!

**Comment by benma** [comment]
utACK

**Pull request merged**